lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -229,13 +244,27 @@ def scrub(transcript)
       }
 
       def add_extra_data(post, payment, options)
-        post[:telephoneNumber] = (options[:billing_address][:phone_number] if options.dig(:billing_address, :phone_number)) || (options[:billing_address][:phone] if options.dig(:billing_address, :phone)) || ''
-        post[:fraudOffset] = options[:fraud_offset] if options[:fraud_offset]
         post[:selectedBrand] = options[:selected_brand] if options[:selected_brand]
         post[:selectedBrand] ||= NETWORK_TOKENIZATION_CARD_SOURCE[payment.source.to_s] if payment.is_a?(NetworkTokenizationCreditCard)
         post[:deliveryDate] = options[:delivery_date] if options[:delivery_date]
         post[:merchantOrderReference] = options[:merchant_order_reference] if options[:merchant_order_reference]
         post[:captureDelayHours] = options[:capture_delay_hours] if options[:capture_delay_hours]
+        post[:deviceFingerprint] = options[:device_fingerprint] if options[:device_fingerprint]
+        post[:shopperIP] = options[:ip] if options[:ip]
+        post[:shopperIP] = options[:shopper_ip] if options[:shopper_ip]
```

Comment:
Are both of these lines meant to assign `shopperIP`? If so, is there a way to make this more concise?

Reply:
Yeah, idk why it was originally done like that. I'll update it

Reply:
Nice! I didn't realize that was existing code that had moved. My bad 🙃